### ISSUE TITLE
Make dependency to lz4 optional

### ISSUE THREAD

**Issue by kazuho** [cybozu/libyrmcds#6]
Now, libyrmcds always links to LZ4, however it is often the case that the application needs to compression at all.

It would be great if the dependency could be optionally turned off at compile time.

**Comment by kazuho** [comment]
https://github.com/kazuho/libyrmcds/commit/b5c741b2e51fb851320afca702e3085c355b17b0 introduces a macro named `LIBYRMCDS_USE_LZ4` which is turned on by default.  Setting the constant to zero the source code no longer includes `lz4.h` nor calls the LZ4 functions.

The commit does not change the Makefile (which always to link to `lz4.o`), for two reasons:
* it does not matter for me, as I will be not using `make` anyways
* basically I am not sure if bundling lz4.o as part of `libyrmcds.a` is a good thing, as lz4.o exports the functions globally - it might cause link issues when an application tries to use both libyrmcds and lz4

**Comment by ymmt2005** [comment]
Agreed!

**Comment by ymmt2005** [comment]
Wondering whether to remove LZ4 support completely, or to make it just optional.

As libyrmcds is meant minimalistic, LZ4 compression can be considered superfluous.
That said, implementing transparent compression on top of libyrmcds is tedious as
one need to wrap many data sending APIs.

One possible solution is to make (de)compression pluggable.  For instance, one might
call `yrmcds_set_compress(compress_func, decompress_func)` to enable transparent
(de)compression.  This, however, imposes a fair amount of work for library users.

Taking these into account, I think it is best at this point to introduce a macro just as 
@kazuho suggested.

**Comment by ymmt2005** [comment]
@kazuho Done in #9.  Please confirm.

**Comment by kazuho** [comment]
Thank you for all the changes.  I no more have any blockers to use the master of libyrmcds; I have successfully switched to using master in https://github.com/h2o/h2o/commit/74a7bd26faa0da4d907e12f9b179ae69267bda6a
